python/cuml/dask/common/input_utils.py

```diff
@@ -215,11 +215,11 @@ def _extract_partitions(dask_obj, client=None):
     # dask.dataframe or dask.array
     if isinstance(dask_obj, dcDataFrame) or \
             isinstance(dask_obj, daskArray):
-        parts = futures_of(client.compute(dask_obj))
+        parts = client.compute(futures_of(dask_obj))
 
     # iterable of dask collections (need to colocate them)
     elif isinstance(dask_obj, Sequence):
-        parts = [futures_of(client.compute(a)) for a in dask_obj]
+        parts = [client.compute(futures_of(d)) for d in dask_obj]
```

Comment:
I modified this slightly in the naive bayes PR to also do the persist here. I've found this alleviates the need to do it in the mixins or model fit functions.